contracts/bridge/src/bridge.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {ERC20} from "lib/openzeppelin-contracts/contracts/token/ERC20/ERC20.sol";
+
+contract Bridge is ERC20 {
+    uint256 public number;
+    uint256 public collateralThreshold;
+    uint256 nextRedeemId;
+    mapping(uint256 => RedeemRequest) public redeemRequests; // cant have struct as key, nor tupple
+    mapping(address => uint256) public suppliedCollateral;
+    uint256 totalCollateral;
+
+    struct RedeemRequest {
+        bool open; // actually redundant now that we have accepterAddress
+        uint256 amountZbtc;
+        uint256 amountBtc;
+        address requesterAddress;
+        address accepterAddress;
+        BitcoinAddress bitcoinAddress;
+    }
+    struct BitcoinAddress {
+        uint256 bitcoinAddress; // todo: use the right type
+    }
+    struct TransactionProof {
+        // todo: fields here
+        uint256 dummy;
+    }
+
+    constructor(uint256 threshold) ERC20("Our ZBTC", "ZBTC") {
+        collateralThreshold = threshold;
+    }
+
+    /// lock COL in exchange for zBTC and cCOL
+    function mintZbtc() public payable {
+        uint256 collateral = msg.value; // this is the amount of eth sent to the contract
+        uint256 mintedAmount = colToBtc(collateral) / collateralThreshold;
+        _mint(msg.sender, mintedAmount);
+
+        suppliedCollateral[msg.sender] += collateral;
+        totalCollateral += collateral;
+    }
+
+    /// request zBTC to be redeemed for given amount of BTC.
+    function requestRedeem(
```

Comment:
I'd like to switch over to a notion o calling it swap instead of redeem. So requestRedeem would be requestSwap, ... and the redeemRequests would be orders placed by users.